CHANGELOG.md

```diff
@@ -7,6 +7,34 @@
 
 - Added a `plainHttp` option to the `v4.Chart` resource. (https://github.com/pulumi/pulumi-kubernetes/issues/3250)
 
+### Changed
+
+- The `pulumi.com/waitFor` annotation now uses an
+  [RFC9535](https://www.rfc-editor.org/rfc/rfc9535.html)-compliant JSONPath
+  parser. This makes it possible to wait for more complex scenarios.
+
+  For example:
+
+  - To wait for a Pod's `status.phase` to be "Running" or "Succeeded":
+
+    `pulumi.com/waitFor: "jsonpath={.status[?@ == 'Running' || @ == 'Succeeded' ]}"`
```

Comment:
Unfortunately this is a little implicit because we can't specify the "phase" field directly https://mailarchive.ietf.org/arch/msg/jsonpath/ueqHXqXa_2hdjIWcjZRKtu3_bRk/